crates/wasm/src/storage.rs

```diff
@@ -36,11 +39,52 @@ pub struct IndexedDBStorage {
 
 impl IndexedDBStorage {
     pub async fn new(constants: IndexedDbConstants) -> WasmResult<Self> {
-        let db_req: OpenDbRequest = IdbDatabase::open_u32(&constants.name, constants.version)?;
+        let mut db_req: OpenDbRequest = IdbDatabase::open_u32(&constants.name, constants.version)?;
+
+        // Conditionally create object stores in the `IdbDatabase` database for testing purposes
```

Comment:
Attempting to migrate this to the test suite has been extremely unsuccessful over the past couple of days. It boils down to **1.** our decision to use `IdbDatabase`, a wrapper for the `IndexDB` database, and **2.** the tight [coupling](https://github.com/penumbra-zone/penumbra/issues/3274) of the `IndexedDBStorage` instance when we initialize the `ViewServer` / `WasmPlanner`. Triggering the creation of an object store in the database requires a an `onupgradeneeded` event, which requires a connection handle to the database instance. Unfortunately, the handle is [consumed](https://github.com/penumbra-zone/penumbra/blob/tx-decompose/crates/wasm/src/storage.rs#L79) in the creation of the database. Anyways, we can't create the database in the test suite directly because any changes made to it won't affect the database instance specific to the `ViewServer` / `WasmPlanner`. 

Dependency injection via a constructor or setter would resolve this, but It doesn't seem to work in this environment since `Serialize` is not implemented for `indexed_db_futures::IdbDatabase`. 

For instance, something like
  ```
  pub async fn set_storage(&mut self, storage: JsValue) -> WasmResult<()>{
        let storage = serde_wasm_bindgen::from_value(storage)?;
        self.storage = Some(storage)
    }
```

would require somehow converting a `JsValue` into `IndexedDBStorage` or vice-versa. 

TLDR; the conditional inside the storage module is the simplest way to mock IndexDB calls and bypass the aforementioned limitations in this environment. I think we can keep it for now.

Reply:
Ah, indeed that is tricky. But if the db instance is expected, we couldn't create the object store? Like this in the test:

```rust
let database: *const IdbDatabase = storage_ref.get_database();
database.create_object_store(...);
```

However, if this is not possible, think it would be worth extracting this logic into a separate method (create_tables_for_test()) so the ::new() method isn't pull of this business logic.

Reply:
This won't work since it'll complain that the creation of an object store requires an `upgradeneeded` event. 

I think a separate method for testing purposes would work well here.